### PR TITLE
Tools for creating Tensile client config files

### PR DESCRIPTION
**Pull request by benjaminulmer**
Using the Tensile client in best-solution mode it a lightweight option for doing GEMM benchmarks without the overhead of rocBLAS. This PR adds some tools for easily creating Tensile client config files to facilitate this process. 

For context, this is to support the work on improving the solution selection in Tensile and addresses [LWPSOLSEL-5](https://ontrack-internal.amd.com/browse/LWPSOLSEL-5).

### TensileCreateLibrary changes
This PR adds two new optional CLI arguments
--build-client: simply builds the Tensile client in the output directory after doing all the normal work
--client-config: outputs a client config file with the library-file and code-object parameters set to the respective files created by the invocation. Also sets best-solution-mode to True

### New TensileClientConfig tool
This tool is meant to provide a quick way to generate a config with the remaining needed parameters set (the ones _not_ set by the --client-config option in CreateLibrary). 

The input to the program is a list of yaml files which need to contain information on the global parameters, a problem definition, and a list of sizes. The exact format of the file(s), however, is a bit flexible. Currently, it can accept a normal Tensile config yaml **or** a set of files being used internally for the solution selection project.

For now, I've made the (potentially controversial choice) to keep the interface very flexible. Instead of specifying what format the input yaml(s) are in, the program does it's best to parse out all the information it can from each file provided based on the formats it knows about. The benefit of this approach is new supported formats can be added and used without having to change how the tool is called. As the code in the solution selection workgroup matures and becomes less volatile, this can be revisited and a more "concrete" interface decided on.

### CODE REVIEW

**Review by lringham — COMMENTED** [review]
LGTM

**Review by zaliu — APPROVED** [review]
LGTM.  @JeremyAdamHart would you please do a quick review of this PR?  Thanks.

**Review by JeremyAdamHart — APPROVED** [review]
Overall looks good to me! Requested changes are minor and shouldn't block merging

File: Tensile/TensileClientConfig.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    sizeList = None
+
+    for config in configs:
+        (myGlobalParams, myProblemDict, mySizeList) = parseConfig(config)
+
+        # if we got data from the config, keep it
+        # if we already had that data, warn and exit
+        if myGlobalParams is not None:
+            if globalParams == {} or globalParams == myGlobalParams:
+                globalParams = myGlobalParams
```
[the remark this answers is not in the document]
Comment (benjaminulmer) [reply Tensile/TensileClientConfig.py:142]: I took a crack at this, but the `elif args.MergeSizes` means this pattern won't work for that block. I'm going to leave it as if for now, but I'll revisit next time I'm working on this file.

**Pull request merged**